### DISCUSSION TITLE
Plugin Model Change - Migrations Not Happening

### DISCUSSION THREAD

**Discussion by cslingerland** [netbox-community/netbox#7316] [Q&A] [answered]
I've got a plugin, and I made some changes to models.py (added a field to a model), but the change is not being detected. I re-installed the plugin with the new field added to models.py, but when I run `python3 manage.py migrate`, it says "No migrations to apply."

The plugin is properly configured as it's working just fine otherwise. Configuration file is done properly.  It's just not picking up my migrations that will be needed.

What am I missing?

(Netbox Version 2.11.12)

**Comment by jeremystretch** [comment, marked as answer]
Try running it with the name of the plugin specified; e.g. `python3 manage.py migrate plugin_name`. I'm not sure why but it seems that Django sometimes doesn't pick up plugin migrations automatically.

**Reply by cslingerland** [reply]
Thanks for the quick reply Jeremy. I tried that as well but it's not picking them up. I'm not familiar with how netbox is picking up migrations from plugins. Could it be something with that mechanism where it's not properly drilling down to check for them? PS. I've restarted the netbox service as well just to cross that off.
(venv) [root@netbox-dev-sf-1 netbox]# python3 manage.py migrate b2b       
CommandError: App 'b2b' does not have migrations.

**Reply by jeremystretch** [reply]
Oh, wait. Are you trying to _apply_ migrations or _create_ them? The later is a different command: `manage.py makemigrations $plugin_name`. This will create the actual migration files on disk, which can then be applied to the database with `manage.py migrate`.

**Reply by cslingerland** [reply]
Sorry for not clarifying. I've just made the changes to the plugin so I need to create them and apply them. I tried makemigrations but it yells at me about being for development only and not helping with missing or unapplied migrations. Is makemigrations disabled by default?

**Reply by jeremystretch** [reply]
Yes, we've disabled the command by default because it can be destructive for users who employ it inadvertently. You'll need to set `DEVELOPER = True` in `configuration.py` first.

**Reply by cslingerland** [reply]
That was it! Apologies if I missed that in the documentation anywhere, but I didn't see it called out. 
Thank you very much
